lib/api/bucketPutVersioning.js

```diff
@@ -80,11 +80,12 @@ function _checkBackendVersioningImplemented(bucket) {
 }
 
 function _isValidReplicationVersioning(result, bucket) {
-    const status = result.VersioningConfiguration.Status[0];
-    // Is there a replication configuration set on the bucket and the user is
-    // attempting to suspend versioning?
-    if (bucket.getReplicationConfiguration()) {
-        return status !== 'Suspended';
+    if (result.VersioningConfiguration.Status) {
```

Comment:
Don't you want to check `result.VersioningConfiguration` as well?

Reply:
I'm assuming it's safe because just after this the same condition is used. However, it's never really a good assumption to make in any case and I agree it's better to be safe as possible.